### PR TITLE
multitools are now gentle on light fixtures

### PR DESCRIPTION
**Pull request by press-mark**
## About The Pull Request
this annoyed me so i fixed it

## How This Contributes To The Skyrat Roleplay Experience
less broken light bulbs after fixing the ballasts

## Changelog

:cl:
qol: reduced the chance of accidentally breaking a light fixture with a multitool when repairing the ballast
/:cl:

### CODE REVIEW

**Comment by theOOZ** [comment]
why not upstream?

File: code/modules/power/lighting/light.dm [hunk 1]
```diff
@@ -325,15 +325,19 @@
...
-	//SKYRAT EDIT ADDITION
-	if(istype(tool, /obj/item/multitool) && constant_flickering)
-		to_chat(user, span_notice("You start repairing the ballast of [src] with [tool]."))
-		if(do_after(user, 2 SECONDS, src))
-			stop_flickering()
-			to_chat(user, span_notice("You repair the ballast of [src]!"))
-		return
+	//SKYRAT EDIT
+	if(istype(tool, /obj/item/multitool))
+		if(constant_flickering)
```
Comment (jjpark-kb) [inline code/modules/power/lighting/light.dm]: early return; so if(!constant_flickering)

File: code/modules/power/lighting/light.dm [hunk 2]
```diff
@@ -325,15 +325,19 @@
...
+			to_chat(user, span_notice("You start repairing the ballast of [src] with [tool]."))
+			if(do_after(user, 2 SECONDS, src))
+				stop_flickering()
+				to_chat(user, span_notice("You repair the ballast of [src]!"))
+			return
+		else
+			to_chat(user, span_warning("[src]'s ballast is already working!"))
+			return
```
Comment (jjpark-kb) [inline code/modules/power/lighting/light.dm]: one indentation backwards

File: code/modules/power/lighting/light.dm [hunk 3]
```diff
@@ -325,15 +325,19 @@
...
-		to_chat(user, span_notice("You start repairing the ballast of [src] with [tool]."))
-		if(do_after(user, 2 SECONDS, src))
-			stop_flickering()
-			to_chat(user, span_notice("You repair the ballast of [src]!"))
-		return
+	//SKYRAT EDIT
+	if(istype(tool, /obj/item/multitool))
+		if(constant_flickering)
+			to_chat(user, span_notice("You start repairing the ballast of [src] with [tool]."))
+			if(do_after(user, 2 SECONDS, src))
```
Comment (jjpark-kb) [inline code/modules/power/lighting/light.dm]: early return

File: code/modules/power/lighting/light.dm [hunk 4]
```diff
@@ -325,15 +325,19 @@
...
+				stop_flickering()
+				to_chat(user, span_notice("You repair the ballast of [src]!"))
+			return
```
Comment (jjpark-kb) [inline code/modules/power/lighting/light.dm]: redundant return if you early returned early as well as moving back the below return one indentation.

File: code/modules/power/lighting/light.dm [hunk 5]
```diff
@@ -325,15 +325,19 @@
...
 		return
 
-	//SKYRAT EDIT ADDITION
-	if(istype(tool, /obj/item/multitool) && constant_flickering)
-		to_chat(user, span_notice("You start repairing the ballast of [src] with [tool]."))
-		if(do_after(user, 2 SECONDS, src))
-			stop_flickering()
-			to_chat(user, span_notice("You repair the ballast of [src]!"))
-		return
+	//SKYRAT EDIT
```
Comment (jjpark-kb) [inline code/modules/power/lighting/light.dm]: Make this modular-- I know you didn't do it, but I would appreciate it.

Comment (press-mark) [reply code/modules/power/lighting/light.dm]: moved it :)

File: code/modules/power/lighting/light.dm (hunk 3 above)
Comment (press-mark) [reply code/modules/power/lighting/light.dm]: not 100% which part u were referring to here

File: code/modules/power/lighting/light.dm (hunk 4 above)
Comment (press-mark) [reply code/modules/power/lighting/light.dm]: kept this one in but moved it down for *visual clarity*

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm [hunk 6]
```diff
@@ -96,3 +96,72 @@
 /obj/item/light/tube
 	icon = 'modular_skyrat/modules/aesthetics/lights/icons/lighting.dmi'
 
+
+// attack with item - insert light (if right type), otherwise try to break the light
```
Comment (jjpark-kb) [inline modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: oh god, this isn't what I meant!

You just need to do . = ..() and then add your new functionality (the skyrat part). I am sure there are examples in the codebase about modularly adding more functionality to a proc.

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm [hunk 7]
```diff
@@ -96,3 +96,72 @@
...
+	if(istype(tool, /obj/item/multitool))
+
+		if(!constant_flickering)
+			to_chat(user, span_warning("[src]'s ballast is already working!"))
+			return
+
+		to_chat(user, span_notice("You start repairing the ballast of [src] with [tool]."))
+		if(do_after(user, 2 SECONDS, src))
+			stop_flickering()
+			to_chat(user, span_notice("You repair [src]'s ballast."))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: if(!do_after) early return

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm [hunk 8]
```diff
@@ -96,3 +96,72 @@
...
+	//Light replacer code
+	if(istype(tool, /obj/item/lightreplacer))
+		var/obj/item/lightreplacer/replacer = tool
+		replacer.ReplaceLight(src, user)
+		return
+
+	if(istype(tool, /obj/item/multitool))
+
+		if(!constant_flickering)
+			to_chat(user, span_warning("[src]'s ballast is already working!"))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: balloon_alert instead "ballast already working!"

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm [hunk 9]
```diff
@@ -96,3 +96,72 @@
...
+			return
+
+		to_chat(user, span_notice("You start repairing the ballast of [src] with [tool]."))
```
Comment (jjpark-kb) [inline modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: balloon_alert instead "repairing ballast..."

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm (hunk 7 above)
Comment (jjpark-kb) [inline modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: balloon_alert instead "ballast repaired!"

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm (hunk 6 above)
Comment (press-mark) [reply modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: ok i think we're on the same page now

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm [hunk 10]
```diff
@@ -96,3 +96,17 @@
 /obj/item/light/tube
 	icon = 'modular_skyrat/modules/aesthetics/lights/icons/lighting.dmi'
 
+
+/obj/machinery/light/attackby(obj/item/tool, mob/living/user, params)
+	if(istype(tool, /obj/item/multitool))
```
Comment (Zonespace27) [inline modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: convert this to multitool_act

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm [hunk 11]
```diff
@@ -96,3 +96,17 @@
...
+
+		if(!constant_flickering)
+			balloon_alert(user, "ballast is already working!")
+			return
+
+		balloon_alert(user, "repairing the ballast...")
+		if(do_after(user, 2 SECONDS, src))
```
Comment (Zonespace27) [inline modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: there's a tool_act proc

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm [hunk 12]
```diff
@@ -96,3 +96,17 @@
...
+			stop_flickering()
+			balloon_alert(user, "ballast repaired!")
+			return
+	. = ..()
```
Comment (Zonespace27) [inline modular_skyrat/modules/aesthetics/lights/code/lighting.dm]:
```suggestion
	return ..()
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm (hunk 11 above)
Comment (press-mark) [reply modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: think i used the right one

File: modular_skyrat/modules/aesthetics/lights/code/lighting.dm (hunk 7 above)
Comment (press-mark) [reply modular_skyrat/modules/aesthetics/lights/code/lighting.dm]: what does this do differently though?

**Comment by jjpark-kb** [comment]
image

If it doesn't work, the whips coming out.

**Pull request merged**